### DISCUSSION TITLE
🚀OAuth2 Operation for Flows - YouTube Uploads, Pinterest Posts etc.

### DISCUSSION THREAD

**Discussion by MMTHD** [directus/directus#21858] [Feature Requests]
Hello everyone, I wanted to ask for your assessment whether it makes sense to build an extension for Directus that provides a new operation for Flows, or if it can be done easier with Directus.

I would think it would be cool if it were possible with Directus Flows to automate uploads to YouTube or other social media channels. However, for YouTube, it's necessary to authorize with the Google API via OAuth2 Flow to create a valid access_token. Unfortunately, this process cannot be easily implemented in Flows because it requires direct user interaction. Therefore, I thought it would be nice to have a custom operation where one can configure an OAuth2 Flow by entering the URIs & credentials required for the OAuth2 Flow and initially run the authorization process after saving the Flows once so that the data such as access_token & refresh_token are stored within the Operation Options. On each Flow run, it will check whether the token is still valid or not and renew it if necessary using the refresh_token.

I tried it, but I lack experience with developing custom operations in Directus. I wanted to use a custom Vue component as an overview operation, where I offer an Authorize button if it hasn't been executed yet and the token along with the refresh token haven't been stored in the operation options. However, since I couldn't pass the ID of the current operation as a parameter in Vue and the documentation is somewhat sparse at that point, I decided to leave it for now.

I think it would be a great enhancement for Directus Flows as it opens up new possibilities and makes it much easier for Directus to interact with other platforms.

From my perspective, it would even be something for the Directus Core.

Maybe I’m missing something obvious here and not seeing how this could be done more easily? Directus already offers various OAuth2/OpenID adapters, but I haven’t been able to retrieve a current token from the currently logged-in user to authenticate myself, for example, with the YouTube API.

What do you think?

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
